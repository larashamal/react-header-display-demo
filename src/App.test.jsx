/**
 * TODO:
 * 1. Write a snapshot test and visually inspect it to insure that it is rendering correctly
 * 2. Get by label text
 * 3. Use `userEvent` to type in the input field and submit the form.
 * 3. Expect the name that you typed in the form `toBeInTheDocument()`.
 */

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import App from "./App";

it("renderers without crashing", () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toMatchSnapshot();
});

// it("calls the submit handler when the form is submitted", () => {
//   const handleSubmit = jest.fn();

//   render(<Form setName={handleSubmit} />);
//   const submitBtn = screen.getByRole("button", { name: /submit/i });

//   userEvent.click(submitBtn);
//   expect(handleSubmit).toHaveBeenCalled();
// });

it("enters name and displays on header", () => {
  const nameInput = "Sara";

  render(<App />);

  const h1 = screen.getByRole("heading", { level: 1 });
  const input = screen.getByLabelText(/name/i);
  const submitBtn = screen.getByRole("button");
  userEvent.type(input, nameInput);
  userEvent.click(submitBtn);

  const h2 = screen.getByRole("heading", { level: 2 });
  screen.getByAltText(nameInput);

  expect(h1.textContent).toContain(nameInput);
  expect(h2.textContent).toContain(nameInput);
});
