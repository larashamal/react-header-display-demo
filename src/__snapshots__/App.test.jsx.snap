// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderers without crashing 1`] = `
Array [
  <h1
    className="my-8 text-center text-3xl"
  >
    Hello 
  </h1>,
  <form
    className="mx-auto flex w-48 flex-col items-center"
    onSubmit={[Function]}
  >
    <div>
      <label
        className="sr-only"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="rounded border pl-2"
        id="name"
        name="formName"
        placeholder="Enter your name"
        type="text"
      />
    </div>
    <button
      className="mt-4 rounded bg-green-500 px-4 py-2 text-white"
      type="submit"
    >
      Submit
    </button>
  </form>,
  <ul>
    <li>
      Football
       - 
      $49.99
    </li>
    <li>
      Baseball
       - 
      $9.99
    </li>
    <li>
      Basketball
       - 
      $29.99
    </li>
    <li>
      iPod Touch
       - 
      $99.99
    </li>
    <li>
      iPhone 5
       - 
      $399.99
    </li>
    <li>
      Nexus 7
       - 
      $199.99
    </li>
  </ul>,
]
`;

exports[`renders correctly 1`] = `null`;
